nncf/quantization/waveq_loss.py

```diff
@@ -0,0 +1,67 @@
+import math
+import torch
+
+from nncf.compression_method_api import CompressionLoss
+from nncf.quantization.layers import QUANTIZATION_MODULES
+from nncf.utils import get_all_modules_by_type
+from nncf.nncf_network import NNCFNetwork
+
+
+class WaveQLoss(CompressionLoss):
+
+    def __init__(self, model: NNCFNetwork):
+        self.model = model
+        # upper line doesn't work
+        # "cannot assign module before Module.__init__() call")
+        self.modules = model.children()
+        self.quantize_modules = self.model_to_quantize_modules_convert(self.model)
```

Comment:
please remove `model`, `modules` and `model_to_quantize_modules_convert` function as not needed